File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/OnlineStatus.java
package l2s.gameserver.network.authcomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.network.authcomm.SendablePacket;

public class OnlineStatus extends SendablePacket {
    private final boolean _online;

    public OnlineStatus(boolean online) {
        _online = online;
    }

    @Override
    protected ByteBuf writeImpl(ByteBuf byteBuf) {
        byteBuf.writeByte(_online ? 1 : 0);
        return byteBuf;
    }

    @Override
    protected byte getOpCode() {
        return 1;
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/OnlineStatus.kt
package l2s.gameserver.network.authcomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.SendablePacket

class OnlineStatus(private val _online: Boolean) : SendablePacket() {

    override val opCode: Byte = 0x01

    override fun writeImpl(byteBuf: ByteBuf): ByteBuf {
        byteBuf.writeByte(if (_online) 1 else 0)
        return byteBuf
    }

}